_posts/vitalii_slobodianyk/2016-11-25-elasticsearch-shading.md

```diff
@@ -0,0 +1,350 @@
+---
+layout: post
+title:  "Shading Elasticsearch"
+breadcrumb: true
+author: vitalii_slobodianyk
+date: 2016-11-25
+categories: vitalii_slobodianyk
+tags:
+    - Elasticsearch
+    - Maven
+teaser:
+    info: A guide how to prepare an Elasticsearch distribution tailored to your needs.
+    image: vitalii_slobodianyk/elastic/elastic-logo.png
+header: 
+    version: small
+    title: Software Engineering Blog
+    image: header-logo-crop.png
+    icon: icon-blog
+---
+
+## Introduction
+
+'Shading' or package renaming(relocation) is a process of creating an [uber-jar](http://imagej.net/Uber-JAR) which contains its dependencies and package names of some of the dependendencies are renamed.
```

Comment:
"a.k.a class relocation". Would have "class relocation" point to https://maven.apache.org/components/plugins/maven-shade-plugin/examples/class-relocation.html